runtime/doc/lsp.txt

```diff
@@ -1109,6 +939,248 @@ with({handler}, {override_config})                            *vim.lsp.with()*
+                                (|vim.ringbuf()|) containing progress messages
+                                sent by the server.
+      • {initialized}           (`true?`)
+      • {workspace_folders}     (`lsp.WorkspaceFolder[]?`) The workspace
+                                folders configured in the client when the
+                                server starts. This property is only available
+                                if the client supports workspace folders. It
+                                can be `null` if the client supports workspace
+                                folders but none are configured.
+      • {root_dir}              (`string`)
+      • {attached_buffers}      (`table<integer,true>`)
+      • {commands}              (`table<string,fun(command: lsp.Command, ctx:
+                                table)>`) Table of command name to function
+                                which is called if any LSP action (code
+                                action, code lenses, ...) triggers the
+                                command. Client commands take precedence over
+                                the global command registry.
+      • {settings}              (`table`)
+      • {flags}                 (`table`)
+      • {get_language_id}       (`fun(bufnr: integer, filetype: string): string`)
+      • {capabilities}          (`lsp.ClientCapabilities`) The capabilities
+                                provided by the client (editor or tool)
+      • {dynamic_capabilities}  (`lsp.DynamicCapabilities`)
+      • {request}               (`fun(method: string, params: table?, handler:
+                                lsp.Handler?, bufnr: integer): boolean,
+                                integer?`) Sends a request to the server. This
+                                is a thin wrapper around {client.rpc.request}
+                                with some additional checking. If {handler} is
+                                not specified, If one is not found there, then
+                                an error will occur. Returns: {status},
+                                {[client_id]}. {status} is a boolean
+                                indicating if the notification was successful.
+                                If it is `false`, then it will always be
+                                `false` (the client has shutdown). If {status}
+                                is `true`, the function returns {request_id}
+                                as the second result. You can use this with
+                                `client.cancel_request(request_id)` to cancel
+                                the request.
+      • {request_sync}          (`fun(method: string, params: table?,
+                                timeout_ms: integer?, bufnr: integer): {err:
+                                lsp.ResponseError?, result:any}?, string?`)
+                                err # a dictionary, where
```

Comment:
👀 

(The vimdoc parser relies on whitespace for tokenization and so can't handle arguments broken over multiple lines.)